extensions/eclipse-che-theia-user-preferences/src/browser/che-shell-layout-restorer.ts

```diff
@@ -0,0 +1,55 @@
+/**********************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+import { FrontendApplication, ShellLayoutRestorer, StorageService, WidgetManager } from '@theia/core/lib/browser';
+import {
+  FrontendApplicationState,
+  FrontendApplicationStateService,
+} from '@theia/core/lib/browser/frontend-application-state';
+import { inject, injectable } from 'inversify';
+
+import { ILogger } from '@theia/core';
+
+@injectable()
+export class CheShellLayoutRestorer extends ShellLayoutRestorer {
```

Comment:
Let's document all the overridden functionality here.
Because it's not obvious why we're patching it downstream instead of providing it upstream.
Several comments will help a lot when we'll need to look at this code in a year.

Reply:
Added documentation section.

Reply:
Thanks!